pkg/provider/device42/device42.go

```diff
@@ -0,0 +1,175 @@
+	errMissingSAK                             = "missing credentials while setting auth"
+)
+
+type Client interface {
+	GetSecret(secretId string) (D42Password, error)
+}
+
+// Device42 Provider struct with reference to a Device42 client.
+type Device42 struct {
+	client Client
+}
+
+func (p *Device42) ValidateStore(esv1beta1.GenericStore) (admission.Warnings, error) {
+	return nil, nil
+}
+
+func (p *Device42) Capabilities() esv1beta1.SecretStoreCapabilities {
+	return esv1beta1.SecretStoreReadOnly
+}
+
+// Client for interacting with kubernetes
+type device42Client struct {
+	kube      kclient.Client
+	store     *esv1beta1.Device42Provider
+	namespace string
+	storeKind string
+}
+type Provider struct{}
+
+func (c *device42Client) getAuth(ctx context.Context) (string, string, error) {
+	credentialsSecret := &corev1.Secret{}
+	credentialsSecretName := c.store.Auth.SecretRef.Credentials.Name
+	if credentialsSecretName == "" {
+		return "", "", fmt.Errorf(errCredSecretName)
+	}
+	objectKey := types.NamespacedName{
+		Name:      credentialsSecretName,
+		Namespace: c.namespace,
+	}
+	// only ClusterStore is allowed to set namespace (and then it's required)
+	if c.storeKind == esv1beta1.ClusterSecretStoreKind {
+		if c.store.Auth.SecretRef.Credentials.Namespace == nil {
+			return "", "", fmt.Errorf(errInvalidClusterStoreMissingSAKNamespace)
+		}
+		objectKey.Namespace = *c.store.Auth.SecretRef.Credentials.Namespace
+	}
+
+	err := c.kube.Get(ctx, objectKey, credentialsSecret)
+	if err != nil {
+		return "", "", fmt.Errorf(errFetchSAKSecret, err)
+	}
+
+	username := credentialsSecret.Data["username"]
+	password := credentialsSecret.Data["password"]
+	if (username == nil) || (len(username) == 0 || password == nil) || (len(password) == 0) {
+		return "", "", fmt.Errorf(errMissingSAK)
+	}
+
+	return string(username), string(password), nil
+}
+
+// NewDevice42Provider returns a reference to a new instance of a 'Device42' struct.
+func NewDevice42Provider() *Device42 {
+	return &Device42{}
+}
+
+func (p *Device42) NewClient(ctx context.Context, store esv1beta1.GenericStore, kube kclient.Client, namespace string) (esv1beta1.SecretsClient, error) {
+	storeSpec := store.GetSpec()
+	if storeSpec == nil || storeSpec.Provider == nil || storeSpec.Provider.Device42 == nil {
+		return nil, fmt.Errorf("no store type or wrong store type")
+	}
+	storeSpecDevice42 := storeSpec.Provider.Device42
+
+	cliStore := device42Client{
+		kube:      kube,
+		store:     storeSpecDevice42,
+		namespace: namespace,
+		storeKind: store.GetObjectKind().GroupVersionKind().Kind,
+	}
+
+	username, password, err := cliStore.getAuth(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Create a new client using credentials and options
+	p.client = NewAPI(ctx, storeSpecDevice42.Host, username, password, "443")
+
+	return p, nil
+}
+
+func (p *Device42) SecretExists(_ context.Context, _ esv1beta1.PushSecretRemoteRef) (bool, error) {
+	return false, fmt.Errorf("not implemented")
```

Comment:
use your notimplemeted error in these places. :)

Reply:
@Skarlso - just made that change, thanks!